### DISCUSSION TITLE
Forbid empty lines at start/end of method

### DISCUSSION THREAD

**Discussion by turbanoff** [checkstyle/checkstyle#11550] [Q&A] [answered]
Is it possible to configure checkstyle to get warnings about empty lines at start/end of methods?
For example I would like to see 2 warnings here (about 2 empty lines),
```
    private double calculate(double a, double b) {

        return a + b;

    }
```

Perhaps later it could be extended to cover all code in brackets. Not only methods.

**Comment by romani** [comment]
not an answer: 
please recheck https://github.com/checkstyle/checkstyle/issues/11286 it might be relevant

answer:
Looks like do not like empty lines in general, can play with some Regexp Checks, try https://checkstyle.org/config_regexp.html#RegexpSinglelineJava with extended list of suppressions by Xpath

if you want to be selective - new Check is required. you can create it https://checkstyle.org/writingchecks.html, there are numerous examples of Checks in our code base. In this project we can host Check that is not well designed but helpful - https://github.com/sevntu-checkstyle/sevntu.checkstyle , this repo has significantly reduced quality level.

**Comment by pbludov** [comment, marked as answer]
Related issue: https://github.com/checkstyle/checkstyle/issues/5439#issuecomment-1097041065

**Reply by rnveach** [reply]
Think https://github.com/checkstyle/checkstyle/issues/5313 is also related
